lib/matobjplist.gi

```diff
@@ -71,33 +71,81 @@ BindGlobal( "MakeIsPlistVectorRep",
         typ := types[1];
     fi;
 
+    if check and ValueOption( "check" ) <> false then
+      if not IsSubset( basedomain, list ) then
+        Error( "the elements in <list> must lie in <basedomain>" );
+      fi;
+    fi;
+
     return Objectify(typ, [ basedomain, list ]);
   end );
 
-InstallMethod( NewVector, "for IsPlistVectorRep, a ring, and a list",
-  [ IsPlistVectorRep, IsRing, IsList ],
-  function( filter, basedomain, l )
-    if ValueOption( "check" ) <> false and not IsSubset( basedomain, l ) then
-      Error( "the elements in <l> must lie in <basedomain>" );
+BindGlobal( "MakeIsPlistMatrixRep",
+  function( basedomain, emptyvector, rowlength, list, check )
+    local fam, types, typ, row;
+    fam:= CollectionsFamily( FamilyObj( basedomain ) );
+
+    # Currently there is no special handling depending on 'basedomain',
+    # the types are always cached in 'fam'.
+    if not IsBound( fam!.PlistMatrixRepTypes ) then
+      # initialize type cache
+      # TODO: make this thread safe for HPC-GAP
+      fam!.PlistMatrixRepTypes:= [
+          NewType( fam, IsPlistMatrixRep ),
+          NewType( fam, IsPlistMatrixRep and IsMutable ),
+      ];
+      fam!.PlistMatrixRepTypesEasyCompare:= [
+          NewType( fam, IsPlistMatrixRep and CanEasilyCompareElements ),
+          NewType( fam, IsPlistMatrixRep and CanEasilyCompareElements and IsMutable ),
+      ];
     fi;
-    return MakeIsPlistVectorRep(basedomain, ShallowCopy(l));
+    if HasCanEasilyCompareElements( Representative( basedomain ) ) and
+       CanEasilyCompareElements( Representative( basedomain ) ) then
+      types:= fam!.PlistMatrixRepTypesEasyCompare;
+    else
+      types:= fam!.PlistMatrixRepTypes;
+    fi;
+    if IsMutable( list ) then
+      typ:= types[2];
+    else
+      typ:= types[1];
+    fi;
+
+    if check and ValueOption( "check" ) <> false then
+      for row in list do
+        if not IsPlistVectorRep( row ) then
+          Error( "the entries of <list> must be in 'IsPlistVectorRep'" );
```

Comment:
Should we have a test case for this? And likewise for the next three errors cases, which according to Codecov are currently not getting exercised?